crates/nargo_cli/src/lib.rs

```diff
@@ -1,5 +1,5 @@
 #![forbid(unsafe_code)]
-#![warn(unused_crate_dependencies, unused_extern_crates)]
```

Comment:
@TomAFrench you were talking about this at the off-site. I found https://github.com/rust-lang/rust/issues/72686 which says that it's so buggy they dropped it from default commands and called it "fixed" but someone commented and said it was still broken.

I removed it because it causes the nix build to fail.